x/group/client/testutil/tx.go

```diff
@@ -2306,7 +2223,213 @@ func (s *IntegrationTestSuite) TestTxLeaveGroup() {
 	}
 }
 
-func (s *IntegrationTestSuite) getGroupIdFromTxResponse(txResp sdk.TxResponse) string {
+func (s *IntegrationTestSuite) TestExecProposalsWhenMemberLeavesOrIsUpdated() {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+
+	weights := []string{"1", "1", "2"}
+	accounts := s.createAccounts(3)
+	testCases := []struct {
+		name      string
+		votes     []string
+		members   []string
+		malleate  func(groupID string) error
+		expectErr bool
+		errMsg    string
+		respType  proto.Message
+	}{
+		{
+			"member leaves while all others vote yes",
+			[]string{"VOTE_OPTION_YES", "VOTE_OPTION_YES", "VOTE_OPTION_YES"},
+			accounts,
+			func(groupID string) error {
+				leavingMemberIdx := 1
+				args := append(
+					[]string{
+						accounts[leavingMemberIdx],
+						groupID,
+
+						fmt.Sprintf("--%s=%s", flags.FlagFrom, accounts[leavingMemberIdx]),
+					},
+					s.commonFlags...,
+				)
+				out, err := cli.ExecTestCLICmd(clientCtx, client.MsgLeaveGroupCmd(), args)
+				s.Require().NoError(err, out.String())
+				var resp sdk.TxResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &resp), out.String())
+				s.Require().Equal(uint32(0), resp.Code, out.String())
+
+				return err
+			},
+			false,
+			"",
+			&sdk.TxResponse{},
+		},
+		{
+			"member leaves while all others vote yes and no",
+			[]string{"VOTE_OPTION_NO", "VOTE_OPTION_YES", "VOTE_OPTION_YES"},
+			accounts,
+			func(groupID string) error {
+				leavingMemberIdx := 1
+				args := append(
+					[]string{
+						accounts[leavingMemberIdx],
+						groupID,
+
+						fmt.Sprintf("--%s=%s", flags.FlagFrom, accounts[leavingMemberIdx]),
+					},
+					s.commonFlags...,
+				)
+				out, err := cli.ExecTestCLICmd(clientCtx, client.MsgLeaveGroupCmd(), args)
+				s.Require().NoError(err, out.String())
+				var resp sdk.TxResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &resp), out.String())
+				s.Require().Equal(uint32(0), resp.Code, out.String())
+
+				return err
+			},
+			true,
+			"PROPOSAL_EXECUTOR_RESULT_NOT_RUN",
+			&sdk.TxResponse{},
+		},
+		{
+			"member that leaves affects the threshold policy outcome",
+			[]string{"VOTE_OPTION_YES", "VOTE_OPTION_NO"},
```

Comment:
I believe for this test case, it would make more sense to have `[]string{"VOTE_OPTION_YES", "VOTE_OPTION_YES"}` which would result in the proposal passing and being executed. Indeed, even though the threshold is 3, since member with index 2 which had a weight of 2 leaves, the total weight of the group is now 2, so if member 0 and member 1 both vote yes, that would result in a tally of 2, meaning the proposal passes while it wouldn't have necessarily passed if member 2 didn't leave

Reply:
I added a test case representing the case you provided. I believe this test is still good to check for edge cases, so I thought, why not have both if that is ok with you?

Reply:
the thing is that in your current test "member that leaves affects the threshold policy outcome" (using `[]string{"VOTE_OPTION_YES", "VOTE_OPTION_NO"}`), if you just don't do what's done in `malleate` (ie member 2 leaving the group), then the test outcome is just the same so I don't believe it's useful for this reason
This doesn't show that a member that leaves affects the threshold policy outcome, it's rather when we have `[]string{"VOTE_OPTION_YES", "VOTE_OPTION_YES"}` that it does

Reply:
see my suggestion in https://github.com/cosmos/cosmos-sdk/pull/11679#discussion_r865638270

Reply:
Ok, will change it!